File: backend/user-service/src/main/java/com/enterprise/user_service/service/UserService.java
package com.enterprise.user_service.service;

import com.enterprise.user_service.dto.request.CreateUserRequest;
import com.enterprise.user_service.dto.request.UpdateUserRequest;
import com.enterprise.user_service.dto.response.UserResponse;

import java.util.List;

public interface UserService {

    UserResponse getUserById(Long id);

    UserResponse getUserByEmail(String email);

    List<UserResponse> getAllUsers();

    UserResponse createUser(CreateUserRequest request);

    UserResponse updateUser(
            Long id,
            UpdateUserRequest request
    );

    UserResponse updateUserByEmail(
            String email,
            UpdateUserRequest request
    );

    void deleteUser(Long id);
}

File: frontend/src/services/userService.js
import api from "./api";

const userService = {
    getProfile: async () => {
        const response = await api.get("/api/v1/users/profile");
        return response.data;
    },

    updateProfile: async (data) => {
        const response = await api.put("/api/v1/users/profile", data);
        return response.data;
    },

    getAllUsers: async () => {
        const response = await api.get("/api/v1/users");
        return response.data;
    },

    getUserById: async (id) => {
        const response = await api.get(`/api/v1/users/${id}`);
        return response.data;
    },

    updateUser: async (id, data) => {
        const response = await api.put(`/api/v1/users/${id}`, data);
        return response.data;
    },

    deleteUser: async (id) => {
        await api.delete(`/api/v1/users/${id}`);
        return true;
    }
};

export default userService;
